### DISCUSSION TITLE
Inconsistent layer order sorting when using nested folders

### DISCUSSION THREAD

**Discussion by LordCemOnur** [nftchef/art-engine#290] [Help + Q&A]
When using nested folders, layers are sometimes ordered wrong. 

Example case; subfolders are named as: 1-Layer, 2-Layer, 3-Layer, etc.
See how they are listed out of order in the DNA:
![image](https://user-images.githubusercontent.com/33794346/158087126-1d128eff-129d-429c-a322-fbb13de04ff1.png)
While the expected outcome would be: 
![image](https://user-images.githubusercontent.com/33794346/158087791-1aa12d47-2a25-4d7d-b076-31fbd4486ec9.png)

This issue does not occur if z indexes are used. Let me know if you need any further details to replicate the issue, and thanks for making such an awesome fork @nftchef!

**Comment by nftchef** [comment]
Thank you! Glad you found z-index.
I believe this is "normal" behavior because of how the generator sorts based on the `leaf` nodes, of the inner-most nested file. Using z-index is the proper solution for fine-grained control here. 

There _could_ be something wrong, but I would need to see how the folders and files are organized and the layersOrder from the config to be able to reproduce.

Thank you

**Comment by LordCemOnur** [comment]
That's what I thought too at first, then I realized the files at the deepest level are not ordered either. Here's my simplified detailed structure:

```
layersOrder: [
      { name: "Body" },
      { name: "VFX" },
      { name: "Background", blend: 'destination-over' },
]
```
Folder structure:
```
> Background
> Body
  > Default
  > Xeneon
    > Clothing-1
       > Xeneon1-1#5.png
       > Xeneon1-2#5.png
       > Xeneon1-3#5.png
       > ...
    > Clothing-2
       > Xeneon2-1#5.png
       > Xeneon2-2#5.png
       > Xeneon2-3#5.png
       > ...
    > Clothing-3
       > Xeneon3-1#5.png
       > Xeneon3-2#5.png
       > Xeneon3-3#5.png
       > ...
    > Clothing-4
       > Xeneon4-1#5.png
       > Xeneon4-2#5.png
       > Xeneon4-3#5.png
       > ...
    > Skin
       > Contessa#5.png
       > Seashell#5.png
       > Whiskey#5.png
       > ...
  > Other...
> VFX
```

**Comment by allendp11** [comment]
> That's what I thought too at first, then I realized the files at the deepest level are not ordered either. Here's my simplified detailed structure:
> 
> ```
> layersOrder: [
>       { name: "Body" },
>       { name: "VFX" },
>       { name: "Background", blend: 'destination-over' },
> ]
> ```
> 
> Folder structure:
> 
> ```
> > Background
> > Body
>   > Default
>   > Xeneon
>     > Clothing-1
>        > Xeneon1-1#5.png
>        > Xeneon1-2#5.png
>        > Xeneon1-3#5.png
>        > ...
>     > Clothing-2
>        > Xeneon2-1#5.png
>        > Xeneon2-2#5.png
>        > Xeneon2-3#5.png
>        > ...
>     > Clothing-3
>        > Xeneon3-1#5.png
>        > Xeneon3-2#5.png
>        > Xeneon3-3#5.png
>        > ...
>     > Clothing-4
>        > Xeneon4-1#5.png
>        > Xeneon4-2#5.png
>        > Xeneon4-3#5.png
>        > ...
>     > Skin
>        > Contessa#5.png
>        > Seashell#5.png
>        > Whiskey#5.png
>        > ...
>   > Other...
> > VFX
> ```

What if I have a folder of gloves? And one is a left handed glove.png, one if a right handed glove.png
But I want the left.png to be over every other layer?
And I want the right to be only over a few layers? How would this be structured?

**Comment by LordCemOnur** [comment]
> What if I have a folder of gloves? And one is a left handed glove.png, one if a right handed glove.png But I want the left.png to be over every other layer? And I want the right to be only over a few layers? How would this be structured?

@allendp11 I'm not sure if it's related with this issue but you'll probably need to use the z-indexing feature for that. While grouping same gloves under a same directory, folder for the left gloves would have highest z-index.

```
> Body
> Gloves
  > Glove A
    > Right
    > z99,Left
  > Glove B
    > Right
    > z99,Left
  > ...
> Other
```
